kafka-client/src/main/java/dev/responsive/kafka/internal/db/mongo/MongoKVTable.java

```diff
@@ -57,17 +63,39 @@ public class MongoKVTable implements RemoteKVTable<WriteModel<KVDoc>> {
 
   private final String name;
 
+  // TODO(sophie): allow use with custom partitioner
+  private Function<Bytes, Integer> partitioner;
+
   private final MongoCollection<KVDoc> docs;
   private final MongoCollection<KVMetadataDoc> metadata;
 
   private final ConcurrentMap<Integer, Long> kafkaPartitionToEpoch = new ConcurrentHashMap<>();
 
+  public static MongoKVTable create(
+      final MongoClient client,
+      final String name,
+      final String changelogTopicName,
+      final Admin admin,
+      final CollectionCreationOptions collectionCreationOptions
+  ) {
+    final int numPartitions = numPartitionsForKafkaTopic(admin, changelogTopicName);
```

Comment:
piping the changelog topic name and the admin all the way down here just to get the number of partitions (which is then used to build a partitioner) seems to be leaking a lot of unnecessary information

instead I suggest that we put the partitioner itself in the `RemoteTableSpec` interface and pass the partitioner directly into this.